server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/MailLoader.java

```diff
@@ -38,18 +42,22 @@
         this.blobIdFactory = blobIdFactory;
     }
 
-    MailWithEnqueueId load(MailReferenceDTO dto) throws MailQueue.MailQueueException {
-        try {
-            MailReference mailReference = dto.toMailReference(blobIdFactory);
-
-            Mail mail = mailReference.getMail();
-            MimeMessage mimeMessage = mimeMessageStore.read(mailReference.getPartsId()).block();
-            mail.setMessage(mimeMessage);
-            return new MailWithEnqueueId(mailReference.getEnqueueId(), mail);
-        } catch (AddressException e) {
-            throw new MailQueue.MailQueueException("Failed to parse mail address", e);
-        } catch (MessagingException e) {
-            throw new MailQueue.MailQueueException("Failed to generate mime message", e);
-        }
+    Mono<MailWithEnqueueId> load(MailReferenceDTO dto) {
+        return Mono.fromCallable(() -> dto.toMailReference(blobIdFactory))
+            .flatMap(mailReference -> buildMail(mailReference)
+                .map(mail -> new MailWithEnqueueId(mailReference.getEnqueueId(), mail)));
+    }
+
+    private Mono<Mail> buildMail(MailReference mailReference) {
+        return mimeMessageStore.read(mailReference.getPartsId())
```

Comment:
can it be empty?

Reply:
No, but it can throw "objNotFound" exceptions